YPP-0023.md

```diff
@@ -0,0 +1,78 @@
+# Proposal
+Enable Yearn USDC Vaults (yvUSDC) as a collateral for USDC loans
+
+# Background
+The yvUSDC token has a $281,774,830 USD market cap and 1,991 holders. The total of value locked in Yearn vaults is over $3.8 billion. Adding yvUSDC would allow us to gauge customer interest and inform our decision on adding other Yearn vault types. Furthermore, the yvUSDC price is based on USDC and we are initially only allow borrowing against it in USDC, so that makes it a lower risk and an acceptable entry point into Yearn vault collateral.
+
+To enable yvUSDC as a collateral we will be using the [pricePerShare](https://github.com/yearn/yearn-vaults/blob/main/contracts/Vault.vy) function on the [yvUSDC token contract](https://etherscan.io/token/0x5f18c75abdae578b483e5f43f12a39cf75b973a9#readContract).  This function calculates the current share value in real time as: ` total vault assets / total token supply`.
+
+# Details
+
+The [addYVUSDC-1.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-1.ts), [addYVUSDC-2.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-2.ts)  and [addYVUSDC-3.ts](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC-3.ts) scripts  will be used, with [this input](https://github.com/yieldprotocol/environments-v2/blob/feat/yvusdc/scripts/governance/addYearnCollateral/yvUSDC/addYVUSDC.mainnet.config.ts):
+
+```
+import { readAddressMappingIfExists } from '../../../../shared/helpers'
+import { USDC, YVUSDC } from '../../../../shared/constants'
+import { YEARN } from '../../../../shared/constants'
+import { FYUSDC2112, FYUSDC2203 } from '../../../../shared/constants'
+
+export const developer = '0xC7aE076086623ecEA2450e364C838916a043F9a8'
+export const deployer = '0xC7aE076086623ecEA2450e364C838916a043F9a8'
+
+export const assets: Map<string, string> = new Map([
+  [USDC,   '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'],
+  [YVUSDC, '0x5f18c75abdae578b483e5f43f12a39cf75b973a9'],
```

Comment:
The YVUSDC address seems correct: it points to "yvUSDC 0.3.0", but i'm not sure how to confirm that is the right address.

Reply:
continued this discussion in slack.  that 0x5f18 is an older version of yvUSDC which uses the 0.3.0 api

we've decided to go with the 0xa354F35829Ae975e850e23e9615b11Da1B3dC4DE (yvUSDC api 0.4.3) per Allan's suggestion